### PR TITLE
Enum cleanup

### PR DESCRIPTION
**Pull request by bkoropoff**
Fixes various warnings caused by enums being mixed up with each other or plain integers.

### CODE REVIEW

File: prboom2/src/dsda/game_controller.c [hunk 1]
```diff
@@ -115,41 +115,47 @@ static void dsda_PollRightStick(void) {
...
+               PollButton(DSDA_CONTROLLER_BUTTON_PADDLE1) |
+               PollButton(DSDA_CONTROLLER_BUTTON_PADDLE2) |
+               PollButton(DSDA_CONTROLLER_BUTTON_PADDLE3) |
+               PollButton(DSDA_CONTROLLER_BUTTON_PADDLE4) |
+               PollButton(DSDA_CONTROLLER_BUTTON_TOUCHPAD);
 
   trigger = dsda_AxisValue(&left_trigger);
   if (trigger)
-    ev.data1.i |= (1 << DSDA_CONTROLLER_BUTTON_TRIGGERLEFT);
+    ev.data1.i |= 1 << DSDA_CONTROLLER_BUTTON_TRIGGERLEFT;
```
Comment (kraflab) [inline prboom2/src/dsda/game_controller.c]: Are these changes related? If not, please remove them, so the PR remains focused.

File: prboom2/src/p_spec.c [hunk 2]
```diff
@@ -6099,25 +6099,25 @@ dboolean P_ExecuteZDoomLineSpecial(int special, int * args, line_t * line, int s
       buttonSuccess = EV_DoZDoomCeiling(ceilLowerByValue, line, args[0],
                                         P_ArgToSpeed(args[1]), 0,
                                         args[2], P_ArgToCrush(args[4]), 0,
-                                        P_ArgToChange(args[3]), false);
+                                        P_ArgToChange(args[3]), crushDoom);
```
Comment (kraflab) [inline prboom2/src/p_spec.c]: `false` and `crushDoom` are very different things, even if they have the same result, so this is going to need a more substantial solution. I'll take a look at this one myself, so you can drop it from this PR.

**Comment by kraflab** [comment]
Could you split this into controller, gl, and map element PRs? I realize that's a bit tedious, but it helps me to digest and merge things in smaller chunks.

File: prboom2/src/p_spec.c (hunk 2 above)
Comment (bkoropoff) [reply prboom2/src/p_spec.c]: `EV_DoZDoomCeiling` explicitly takes a `crushmode_e`.  I dropped these changes from the PR.

**Comment by bkoropoff** [comment]
This PR is very small now, so I'm not sure it needs to be split any more, but let me know.

**Comment by kraflab** [comment]
👍

File: prboom2/src/p_spec.c (hunk 2 above)
Comment (kraflab) [reply prboom2/src/p_spec.c]: I went into the zdoom source and it uses `crushDoom` here as well, so `false` is actually a mistake in the first place. Maybe this predates the enum. If you could open a PR adding the changes back, I would appreciate it 😸
